backend/controllers/appointments/addAppointment.js

```diff
@@ -1,10 +1,91 @@
+const moment = require('moment')
+
 const {
   app: { notmEndpointUrl: dpsUrl },
 } = require('../../config')
-const { properCaseName } = require('../../utils')
-const { buildDateTime, DATE_TIME_FORMAT_SPEC } = require('../../../src/dateHelpers')
+const { properCaseName, formatName } = require('../../utils')
+const { buildDateTime, DATE_TIME_FORMAT_SPEC, DAY_MONTH_YEAR } = require('../../../src/dateHelpers')
 const { serviceUnavailableMessage } = require('../../common-messages')
-const { repeatTypes, getValidationMessages, endRecurringEndingDate } = require('../../shared/appointmentConstants')
+const { repeatTypes, endRecurringEndingDate, validateComments } = require('../../shared/appointmentConstants')
+
+const validateDate = (date, errors) => {
+  const now = moment()
+  if (!date) errors.push({ text: 'Select the appointment date', href: '#date' })
+
+  if (date && !moment(date, DAY_MONTH_YEAR).isValid())
+    errors.push({ text: 'Enter the date in DD/MM/YYYY format', href: '#date' })
+
+  if (date && moment(date, DAY_MONTH_YEAR).isBefore(now, 'day'))
+    errors.push({ text: 'Select a date that is not in the past', href: '#date' })
```

Comment:
Not 100% sure I remember how this works, but is this comparing just the day? Will it fail if for example today is 14th Aug and I tried to put in an appointment for 11th Sep?

Reply:
Not sure, it's been working so far. I just pulled it from the validation common module.